src/handlers/records-write.ts

```diff
@@ -50,20 +50,25 @@ export class RecordsWriteHandler implements MethodHandler {
       recordId  : message.recordId
     };
     const existingMessages = await this.messageStore.query(tenant, query) as (RecordsWriteMessage|RecordsDeleteMessage)[];
+    const newestExistingMessage = await Message.getNewestMessage(existingMessages);
 
     // if the incoming write is not the initial write, then it must not modify any immutable properties defined by the initial write
     const newMessageIsInitialWrite = await recordsWrite.isInitialWrite();
     if (!newMessageIsInitialWrite) {
+      const initialWrite = await RecordsWrite.getInitialWrite(existingMessages);
+      if (initialWrite === undefined || newestExistingMessage?.descriptor.method === DwnMethodName.Delete) {
```

Comment:
Returning "initial write is not found" when latest existing message is a delete seems to be misleading?

Is it possible to give specific error? Maybe with a check for the "latest message being a delete" after line 53?